src/flagStore.ts

```diff
@@ -43,48 +42,36 @@ export class FlagStore {
 		}
 	}, 200);
 
-	async start() {
+	start(): LaunchDarkly.LDClient {
 		if (!['accessToken', 'baseUri', 'streamUri', 'project', 'env'].every(o => !!this.config[o])) {
 			console.warn('LaunchDarkly extension is not configured. Language support is unavailable.');
 			return;
 		}
 
-		const sdkKey = await this.getLatestSDKKey();
-		if (!sdkKey) {
-			return;
-		}
+		var sdkKey: string;
 
-		const ldConfig = this.ldConfig();
-		this.updateProcessor = StreamProcessor(sdkKey, ldConfig, Requestor(sdkKey, ldConfig));
-		return new Promise((resolve, reject) => {
-			this.updateProcessor.start(err => {
-				if (err) {
-					let errMsg: string;
-					if (err.message) {
-						errMsg = `Error retrieving feature flags: ${err.message}.`;
-					} else {
-						console.error(err);
-						errMsg = `Unexpected error retrieving flags.`;
-					}
-					reject(errMsg);
-					return;
-				}
-				resolve();
-				process.nextTick(function() {});
+		this.getLatestSDKKey().then(sdkKey => {
```

Comment:
Related to my other comment about the event listener, we could keep `flagStore.start` as it was originally, storing the ldClient internally-- exposing `.on` as a wrapped event listener

Reply:
sounds good, I'll look to implement this.